### DISCUSSION TITLE
Mesures incomplètes et accompagnement

### DISCUSSION THREAD

**Discussion by Hant65** [Jean-Romain/tabliato#38] [General]
Bonjour, 

J'ai du mal avec le codage des mesures incomplètes, et de leur accompagnement. J'ai bien vu dans le manuel Tabliato l'instruction `\partial` à faire suivre par [nombre de notes] * [durée].  (NB :  dans les "exemples" intégrés à Tabliato,  l'ordre est inverse : [durée] *  [nombre de notes] : est-ce vraiment équivalent ?).

Mais surtout : Comment faire lorsque les notes de cette mesure incomplète ont des durées différentes ? 

Ensuite : Comment faire lorsque cette mesure incomplète en début de morceau doit être rejouée lors de la reprise, car elle constitue le complément de la mesure (également incomplète) de la fin du morceau ?
Accessoirement : comment coder un instruction de type "Da Capo" 

Pour expliciter, je joins la tablature annotée et le .dtb du célèbre "Inconnu de Limoise". Merci de vos conseils ! 

[InconnuLimoise_Essai.zip](https://github.com/Jean-Romain/tabliato/files/10725852/InconnuLimoise_Essai.zip)

**Comment by Jean-Romain** [comment]
Ce qu'il faut c'est que la durée totale soit égale à l'anacrouse

- Anacrouse sur 1 noire (donc deux croches) : ` \partial 4`
- Anacrouse sur 3 croches (1 noires + 1 croche) `\partial 3*8` (peut être que `\partial 4.` fonctionne).

En fait je ne suis moi même pas toujours sûr de ce qui marche ou ne marche pas. Par ex. `\partial 2*8` semble ne pas marcher. Bref il faut essayer mais l'idée c'est qu'il doit y avoir la durée totale.

Je n'ai pas testé mais ca devrait ressembler à ca

```r
\partial 2*4 p6:8 p6':4 p8:8 [r:8 g:4.]
```

Pour le reste ca resemble à un gros bug. je vais tester plus tard.

**Reply by Hant65** [reply]
Sans l'accompagnement, `\partial 2 p6:8 p6':4 p8:8` fonctionne (de fait, on a bien deux temps dans cette première mesure). Mais pas `\partial 2*4 p6:8 p6':4 p8:8`, que vous m'aviez indiqué, et qui me renvoie une "erreur inconnue, le document n'a pas été compilé, code probablement incorrect". 
Pour l'accompagnement, le motif  `B a a` du menu déroulant n'est pas accepté. Parce qu'on est en 9/8 ? J'ai donc introduit `\motif [B:4. a:4. a:4.]` en début de code. L'accompagnement tombe alors bien sur les notes souhaitées, mais c'est beaucoup trop "traînant". Et ça ne va pas si je réduis la durée des basses et accords du `\motif`. 
Pour la dernière mesure, également incomplète (2 temps et demi), et qui fait partie d'une alternative, je n'ai pas encore réussi à introduire  `\partial` au bon endroit à l'intérieur des accolades.

**Reply by Hant65** [reply]
Bon, j'ai fait ça (ci dessous) qui va presque bien. En fait, pas besoin de déclarer la dernière mesure comme incomplète. 
Reste à voir : réduire la durée des notes de l'accompagnement, et renvoyer au début pour la reprise. Les lignes de portée sont revenues, elles avaient peut être  été effrayées par les erreurs de code ? 
Bonne fin de soirée . 

```
\motif [B:4. a:4. a:4.] 
\partial 2 p6:8 p6':4 p8:8 [r:8 g:4.]
p7':4. ~ p7':4 p6':8 t8':4 t7':8 C
t9:4. ~ t9:4 t9:8 t8':4 p8':8 A
t8':4. ~ t8':4 t9:8 t7':4 t8:8 F
\break
p6:4. ~ p6:4 p6:8 p6':4 p8:8 G
p7':4. ~ p7':4 p6':8 t8':4 t7':8 [C c g]
t9:4. ~ t9:4 t9:8 t8':4 p8':8 A
t8':4. ~ t8':4 t9:8 t7':4 t8:8 F
t7':4. ~ t7':4 t9:8 t8':4 p8':8 G
\break
\repeat volta 2
{
t9':4. p9':4 t10':8 t9':4 t10':8 F
p8':4. p8':4 t8':8 p7':4 p8:8 C
t8:4 t6':8 t8:4 t7':8 t9:4 t8:8 A
p6:2. p6':4 p7:8 E
t7:4. t7:4 t6':8 t8:4 t7':8 F
t8:4. t6':4. t7:4. F
p7:4. p7:4 p6':8 p8:4 p7':8 G
}
\alternative
{
	{p8:4. p8:4 p7':8 t8':4 p8':8 G}
	{p6':4. ~ p6:4 [C:4. c:4. r:4.]}
}
```

**Reply by Jean-Romain** [reply]
`B a a`  fonctionne en 9/8. Peut être un bug mais moi je peux choisir `B a a`. J'écrirais

```
\partial 2 p6:8 p6':4 p8:8 [r:2]

p7':4. ~ p7':4 p6':8 t8':4 t7':8 C
t9:4. ~ t9:4 t9:8 t8':4 p8':8 A
t8':4. ~ t8':4 t9:8 t7':4 t8:8 F

\break

p6:4. ~ p6:4 p6:8 p6':4 p8:8 G
p7':4. ~ p7':4 p6':8 t8':4 t7':8 [C c g]
t9:4. ~ t9:4 t9:8 t8':4 p8':8 A
t8':4. ~ t8':4 t9:8 t7':4 t8:8 F
t7':4. ~ t7':4 t9:8 t8':4 p8':8 G

\break

\repeat volta 2
{
    t9':4. p9':4 t10':8 t9':4 t10':8 F
    p8':4. p8':4 t8':8 p7':4 p8:8 C
    t8:4 t6':8 t8:4 t7':8 t9:4 t8:8 A
    p6:2. p6':4 p7:8 E
    t7:4. t7:4 t6':8 t8:4 t7':8 F
    t8:4. t6':4. t7:4. F
    p7:4. p7:4 p6':8 p8:4 p7':8 G
}
\alternative
{
   {p8:4. p8:4 p7':8 t8':4 p8':8 G}
   {p6':4. ~ p6':4 [C:4. r:4]}
}

\bar "|." 
```

Attention c'est un `6'` à la fin

**Reply by Hant65** [reply]
Merci de votre aide. Je confirme que dès que je déclare la mesure en 9/8, je ne peux plus choisir le motif "B a a " (il disparaît de la liste déroulante, où il ne reste plus qu'à choisir entre "Manuel" et "B aB aB a"). Je choisis donc "Manuel", et je définis  \motif [B:4. a:4. a:4.], qui malheureusement prolonge beaucoup trop le son des notes de l'accompagnement. Comment restaurer une liste déroulante complète ? Ou sinon, comment définir un motif correspondant à ce qui a été défini pour le "B a a " ?
Par ailleurs, le \bar "|." ne génère hélas pas un retour au tout début du morceau dans le fichier mid (ce qui n'est pas bien grave), mais surtout ne correspond pas au "Da Capo" qu'il faudrait inscrire sur la partition.  Je n'ai encore rien trouvé d'explicite sur ce sujet dans Lilypond. A suivre ...

**Reply by Jean-Romain** [reply]
`\motif [B:8. r:8. a:8. r:8. a:8. r:8.]` fera l'affaire en attendant que je trouve pourquoi des éléments de la liste disparaissent. Je suppose que vous utilisez la version Window?

> Par ailleurs, le `\bar "|."` ne génère hélas pas un retour au tout début du morceau dans le fichier mid 

Non c'est une barre de fin. Je n'ai jamais vu aucune tablature s'embêter avec des notation *da capo*. En vrai en trad toutes les pieces sont implicitement *da capo*. D'après la documentation il faudrait utiliser `\repeat segno`. En l'état c'est impossible car tabliato interprète mal cette commande et bloque. Il faudrait corriger le bug.

**Reply by Hant65** [reply]
Oui, j'utilise la  version Windows. Et très bien pour le nouveau motif. Bien d'accord avec vous, je ne m'obstinerai pas sur le Da Capo. Merci !
